packages/expo-image-manipulator/ios/ImageManipulations.swift

```diff
@@ -0,0 +1,211 @@
+    targetSize.width = width
+    targetSize.height = width / imageRatio
+  }
+  if let height = resize.height {
+    targetSize.height = height
+    targetSize.width = targetSize.width == 0 ? imageRatio * targetSize.height : targetSize.width
+  }
+
+  UIGraphicsBeginImageContextWithOptions(targetSize, false, 1.0)
+  image.draw(in: CGRect(origin: .zero, size: targetSize))
+
+  guard let newImage = UIGraphicsGetImageFromCurrentImageContext() else {
+    UIGraphicsEndImageContext()
+    throw NoImageInContextError()
+  }
+
+  UIGraphicsEndImageContext()
+  return newImage
+}
+
+/**
+ Creates a new image by rotating given image by the rotate angle.
+ */
+internal func manipulate(image: UIImage, rotate: Double) throws -> UIImage {
+  guard let cgImage = image.cgImage else {
+    throw ImageNotFoundError()
+  }
+  let rads = rotate * Double.pi / 180
+  let rotatedView = UIView(frame: CGRect(origin: .zero, size: image.size))
+
+  rotatedView.transform = CGAffineTransform(rotationAngle: rads)
+
+  let rotatedSize = rotatedView.frame.size
+  let origin = CGPoint(x: -image.size.width / 2, y: -image.size.height / 2)
+
+  return try drawInNewContext(size: rotatedSize) { context in
+    context.translateBy(x: rotatedSize.width / 2, y: rotatedSize.height / 2)
+    context.rotate(by: rads)
+    context.scaleBy(x: 1.0, y: -1.0)
+    context.draw(cgImage, in: CGRect(origin: origin, size: image.size))
+  }
+}
+
+/**
+ Creates a new image by flipping given image vertically or horizontally.
+ */
+internal func manipulate(image: UIImage, flip: FlipType) throws -> UIImage {
+  let imageView = UIImageView(image: image)
+
+  return try drawInNewContext(size: imageView.frame.size) { context in
+    switch flip {
+    case .vertical:
+      let transform = CGAffineTransform(a: 1, b: 0, c: 0, d: -1, tx: 0, ty: imageView.frame.size.height)
+      context.concatenate(transform)
+    case .horizontal:
+      let transform = CGAffineTransform(a: -1, b: 0, c: 0, d: 1, tx: imageView.frame.size.width, ty: 0)
+      context.concatenate(transform)
+    }
+    imageView.layer.render(in: context)
+  }
+}
+
+/**
+ Creates a new image by cropping given image to specified width and height.
+ */
+internal func manipulate(image: UIImage, crop: CropRect) throws -> UIImage {
+  let rect = crop.toRect()
+  let isOutOfBounds = rect.origin.x > image.size.width
+  || rect.origin.y > image.size.height
+  || rect.width > image.size.width
+  || rect.height > image.size.height
+
+  guard !isOutOfBounds else {
+    throw ImageInvalidCropError()
+  }
+  guard let cgImage = image.cgImage?.cropping(to: rect) else {
+    throw ImageCropFailedError(rect: rect)
+  }
+  return UIImage(cgImage: cgImage, scale: image.scale, orientation: image.imageOrientation)
+}
+
+/**
+ Makes sure the image is oriented up and not mirrored.
+ Guarantees that the original pixel data matches the displayed orientation.
+ */
+internal func fixImageOrientation(_ image: UIImage) throws -> UIImage {
```

Comment:
**⚠️  SwiftLint Violation: Cyclomatic Complexity**
Function should have complexity 10 or less: currently complexity equals 13